docs/operating-eck/eck-permissions.asciidoc

```diff
@@ -0,0 +1,112 @@
+
+[id="{p}-{page_id}"]
+= ECK permissions
+
+Installing and running ECK, as well as using ECK-managed resources requires certain Kubernetes link:https://kubernetes.io/docs/reference/access-authn-authz/rbac[permissions]. Each of those permissions is listed below, together with a reason why it is needed.
+
+
+[float]
+[id="{p}-{page_id}-defaults"]
+== Installing CRDs
+
+Below is the permission required to install CRDs. CRDs (link:https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/[CustomResourceDefinitions]) are the only non-namespaced resources required to be installed.
+
+[width="100%",cols=".^35m,.^25m,.^5d,.^35",options="header"]
+|===
+|Name|API group|Optional?|Usage
+|CustomResourceDefinition|apiextensions.k8s.io|no|Extend Kubernetes APIs with Elastic Stack applications resources
+|===
+
+[float]
+[id="{p}-{page_id}-installing"]
+== Installing the ECK operator
+
+Below are the permissions required to install ECK operator in a Kubernetes cluster.
+
+[width="100%",cols=".^1m,.^1m,.^1d,.^97",options="header"]
+|===
+|Name|API group|Optional?|Usage
+|StatefulSet or Deployment|apps|no|ECK operator `Pod` deployment method. By default, `Deployment` is used only in OpenShift.
+|ServiceAccount|core|no|Service account that the operator Pods run as
+|Role or ClusterRole|rbac.authorization.k8s.io|no|Role bound to the operators Service account. Depending on the installation type (global/restricted) either global (ClusterRole) or namespaced (Role) resource is needed.
+|RoleBinding or ClusterRoleBinding|rbac.authorization.k8s.io|no|Binding between the operators role and the operators service account. Depending on installation type (global/restricted) either global (ClusterRoleBinding) or namespaced (RoleBinding) resource is needed.
+|ConfigMap|core|yes|Configuration parameters of the Operator. They can be specified directly in the StatefulSet (or Deployment) resource instead.
+|Namespace|core|yes|Namespace where the Operator will run. It can be a pre-existing namespace as well.
+|ValidatingWebhookConfiguration|admissionregistration.k8s.io|yes|Validating webhook installation. It provides fast feedback for the user directly as a APIServer response. Very similar set of validations is done by the operator itself, but the results are only available through operator logs. See link:https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-webhook.html[docs] for more.
+|Secret|core|yes|Secret containing validating webhooks endpoint CA certificate
+|Service|core|yes|Service for validating webhook endpoint
+|===
+
+And all permissions that <<{p}-{page_id}-running>> section specifies.
+
+[float]
+[id="{p}-{page_id}-running"]
+== Running ECK operator
+
+Below are the permissions that are needed by the Service Account that ECK Operator runs as.
+
+[width="100%",cols=".^1m,.^1m,.^1d,.^97",options="header"]
+|===
+|Name|API group|Optional?|Usage
+|Pod||no|Assuring expected Pods presence during Elasticsearch reconciliation +
```

Comment:
Maybe just: The operator needs to list, get and delete Pods during Elasticsearch reconciliation?

Reply:
I wanted to go one level deeper with those descriptions and inform why exactly we need those. It's not obvious given that we operate on StatefulSets, Deployment and DaemonSet. What verbs are needed for which resources is something that is already there in Role spec, but what they are used for is what I think we are missing right now.

Reply:
Yeah that's fine, I was just struggling with the wording a bit. Also you only mention "assuring presence" but we also actively delete Pods.

Reply:
Sound good. Gets (lists) and creates are explained, I've added deletes as well.